### ISSUE TITLE
paraglide-js not working

### ISSUE THREAD

**Issue by PatrykWalach** [oven-sh/bun#12777]
### What version of Bun is running?

1.1.20+ae1948925

### What platform is your computer?

Microsoft Windows NT 10.0.22631.0 x64

### What steps can reproduce the bug?

```sh
bun init
bunx @inlang/paraglide-js@latest init
en // or any language
bun run --bun build
```
I've also created a [repo](https://github.com/PatrykWalach/bunjs-paraglide)

`"@inlang/paraglide-js": "1.11.2"`

### What is the expected behavior?

Running with node works `bun run build`
would expect for this to work in bun too

expected out:
```
$ paraglide-js compile --project ./project.inlang --outdir ./src\paraglide
ℹ [paraglide] Compiling inlang project at "./project.inlang".
Using existing cloned repo
ℹ [paraglide] Successfully compiled the project.
```

### What do you see instead?

I get the error instead:
```
$ paraglide-js compile --project ./project.inlang --outdir ./src\paraglide
ℹ [paraglide] Compiling inlang project at "./project.inlang".
ENOENT: No such file or directory
   errno: -2
 syscall: "stat"


Bun v1.1.20 (Windows x64)
error: script "build" exited with code 1
```

**Comment by frankmayer** [comment]
FYI: Same command on linux  (inside a bun workspace) runs, but throws a warning:
` WARN  Could not find repository root for path /somepath/apps/my-app/project.inlang`

But in a basic **SvelteKit** project (inside a bun workspace), having installed paraglide according to its instructions, running 

`bun --bun run dev`
throws:

```
ENOENT: No such file or directory
   errno: -2
 syscall: "stat"

```
(same as above)

while
` bun run dev` (node mode)
starts the dev environment normally

Not sure, if it's triggered by the same issue, but I thought it's worth mentioning.

**Comment by Droidion** [comment]
Still a problem on bun 1.1.33

**Comment by tobyspark** [comment]
Like above, the Sveltekit + bun workflow fails for me on including paraglide, with the uninformative `ENOENT`. I’ve dug into paraglide, and there’s a lot of filesystem code in there, including a `nodeishFS` wrapper and `fs.stat` calls... but some naive logging hasn’t hit the code path that’s failing. (Bun 1.1.34, Paraglide 1.11.3)

But for _Bun_, what I don’t understand is why I seemingly can’t get debug info from the crash. I can’t find any verbose flags, and e.g. `bun --bun --inspect-brk dev` is not waiting for a debugger to attach.

ps. On Apple Silicon I can neither run the SvelteKit + Bun example with node (`bun dev` → `Cannot find module @rollup/rollup-darwin-x64`) nor Bun (`bun --bun dev`, this issue).

**Comment by samuelstroschein** [comment]
Adding context to the filesystem calls in paraglide js: 

- paraglide v1 uses ["git as a backend"](https://github.com/orgs/opral/discussions/115)  (mostly for apps like [Fink](https://inlang.com/m/tdozzpar/app-inlang-finkLocalizationEditor) or [Sherlock](https://inlang.com/m/r7kp499g/app-inlang-ideExtension))
- the filesystem calls are wrapped in a nodeish like fs interface to enable in memory fs implementations in the browser
- i assume the wrapping might be the issue
- the wrapping is gone in paraglide js v2 which will be released sometime early 2025

**Comment by lesjul** [comment]
@samuelstroschein I'd like to try it out on my SvelteKit app. Could you provide a documentation to migrate into v2 with SvelteKit please ?

**Comment by samuelstroschein** [comment]
I will update the SvelteKit adapter this week. Migration should be non-breaking.

**Comment by PatrykWalach** [comment]
`@inlang/paraglide-js@2.0.0-beta.12` works fine in node

In `bun@1.2` or `bun@1.1` it shows `[paraglide] Successfully compiled the project.`
but the generated `messages.js` is empty

**Comment by lesjul** [comment]
@samuelstroschein Hello, have you updated the sveltekit-adapter ?

**Comment by oskar-gmerek** [comment]
@lesjul https://github.com/opral/monorepo/tree/main/inlang/packages/paraglide/paraglide-sveltekit

**Comment by lesjul** [comment]
@oskar-gmerek Have you tried yet on Sveltekit ?

I cannot find paraglideAdapter when I tried to import it.

`import * as paraglideAdapter from '$lib/paraglide/adapter';`

**Comment by greybluesea** [comment]
> [@lesjul](https://github.com/lesjul) https://github.com/opral/monorepo/tree/main/inlang/packages/paraglide/paraglide-sveltekit

I'm using bun,  it still says: (ts: Module '"@inlang/paraglide-sveltekit"' has no exported member 'createI18n'), and not working with 'setLocale'. Thanks

**Comment by lesjul** [comment]
@oskar-gmerek 
I ran `bun add @inlang/paraglide-sveltekit@beta` then `bun dev`

and I get this error:

![Image](https://github.com/user-attachments/assets/30d83d33-11d1-4d15-a851-b3f9ff47dbad)

**Comment by oskar-gmerek** [comment]
@lesjul Remove node_modules, .svelte-kit, build folders and try again.

As you can see in the error, there is complain about `paraglideVitePlugin` that is not exported from paraglide-js. It must be in cache, as you clearly do not import it in your `vite.config.ts`.(if the screenshot showing whole file)

**Comment by lesjul** [comment]
@oskar-gmerek I still get this error :(

**Comment by oskar-gmerek** [comment]
@lesjul try to remove 'bun.lock'/'bun.lockb' as well as 'node_modules' and try again.

**Comment by samuelstroschein** [comment]
@lesjul did you update @inlang/paraglide-js to a v2 beta version as well? you might be running on v1. (the adapter, and paraglide js are updated independently)

![Image](https://github.com/user-attachments/assets/2c201fec-9c8c-415d-a9e2-417e0150f57d)

**Comment by samuelstroschein** [comment]
Works with Paraglide JS 2.0-beta.16. I think this can be closed :)

![Image](https://github.com/user-attachments/assets/c479f70f-ac32-49e3-8034-79f5f8c2c397)

**Comment by PatrykWalach** [comment]
Works for me as well with bun 1.2.2 and paraglide 2.0.0-beta.16

**Comment by lesjul** [comment]
@samuelstroschein has the documentation been updated to migrate for v2 ?

**Comment by samuelstroschein** [comment]
@lesjul no because paraglide js 2.0's API is still changing. albeit i expect https://github.com/opral/inlang-paraglide-js/issues/378 to be the last API change

**Comment by samuelstroschein** [comment]
@lesjul by the end of this month (February)

**Comment by alec-c4** [comment]
@samuelstroschein hi! Will it fix similar issue with `@inlang/paraglide-sveltekit`?

**Comment by samuelstroschein** [comment]
@lesjul yes you can use the latest paraglide js 2.0 beta. i am squashing a few bugs here and there but overall its stable https://inlang.com/m/gerre34r/library-inlang-paraglideJs
